packages/dara-core/dara/core/configuration.py

```diff
@@ -52,6 +52,7 @@ class Configuration(GenericModel):
     """Definition of the main framework configuration"""
 
     auth_config: BaseAuthConfig
+    registry_lookup: Dict[str, Callable]
```

Comment:
Also let's update the type everywhere to specify the Callable must return a coroutine, if when we call it in RegistryLookup we always `await` it. This way users know they have to provide an `async` function